### PR TITLE
Some corrections

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

**Comment by lovettchris** [comment]
Thanks, this looks great, but I have some questions about the BeforeScriptCOmmandStackCallback, looks like the typedef does return std:;string.

typedef std::function<std::string (const ShellCommandParameters& params, std::string command_line)> BeforeScriptCommandStartCallback;

**Review by bapan0000 — APPROVED** [review]

**Comment by sytelus** [comment]
Looks like @lovettchris had already made changes to compile against GCC 6 and the build was successful. Could you please provide more details such as error messages that required these changes? We would like to know what exact error/warnings each change is resolving.

File: AirLib/deps/rpclib/include/rpc/msgpack/vrefbuffer.hpp [hunk 1]
```diff
@@ -202,7 +202,7 @@ class vrefbuffer {
         if(to->m_tail + nused < m_end) {
             const size_t tosize = to->m_tail - to->m_array;
             const size_t reqsize = nused + tosize;
-            size_t nnext = (to->m_end - to->m_array) * 2;
+            size_t nnext = (size_t)(to->m_end - to->m_array) * 2;
```
Comment (patrickelectric) [inline AirLib/deps/rpclib/include/rpc/msgpack/vrefbuffer.hpp]: error: assuming signed overflow does not occur when simplifying multiplication [-Werror=strict-overflow]

File: AirLib/src/control/DroneControlBase.cpp [hunk 2]
```diff
@@ -8,6 +8,10 @@
 #include "AirSim.h"
 #endif
 
+#ifndef FLT_MAX
+#define FLT_MAX     __FLT_MAX__
+#endif
+
```
Comment (patrickelectric) [inline AirLib/src/control/DroneControlBase.cpp]: Solve: `FLT_MAX not declared`

File: MavLinkCom/MavLinkTest/main.cpp [hunk 3]
```diff
@@ -694,11 +694,11 @@ std::vector<std::string> parseArgs(std::string s)
...
 
-void HexDump(uint8_t *buffer, int len)
+void HexDump(uint8_t *buffer, uint len)
 {
-	for (int i = 0; i < len; i += 16) {
+	for (uint i = 0; i < len; i += 16) {
 
-		int j = 0;
+		uint j = 0;
 		for (j = i; j < i + 16 && j < len; j++)
```
Comment (patrickelectric) [inline MavLinkCom/MavLinkTest/main.cpp]: Solve: `error: assuming signed overflow does not occur when reducing constant in comparison [-Werror=strict-overflow]`

File: DroneShell/include/linenoise.hpp [hunk 4]
```diff
@@ -1511,7 +1511,7 @@ inline size_t unicodeColumnPosForMultiLine(char* buf, size_t buf_len, size_t pos
 
     size_t off = 0;
     while (off < buf_len) {
-        int cp;
+        int cp = 0;
```
Comment (patrickelectric) [inline DroneShell/include/linenoise.hpp]: Solve: `error: 'cp' may be used uninitialized in this function [-Werror=maybe-uninitialized]`

**Comment by patrickelectric** [comment]
@sytelus Done.

**Comment by patrickelectric** [comment]
@sytelus @lovettchris  Sorry, almost everything can be solved with StrictMode.

**Comment by sytelus** [comment]
Merged. FYI - I'd to add `typedef unsigned int uint` for Windows build.

**Pull request merged**